watertap/unit_models/zero_order/pump_variable_zo.py

```diff
@@ -0,0 +1,133 @@
+
+        self.flow_bep = Var(
+            units=pyunits.m**3 / pyunits.s, doc="Best efficiency point flowrate"
+        )
+
+        self.flow_ratio = Var(
+            self.flowsheet().config.time,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Ratio between instantaneous flowrate and best efficiency point flowrate",
+        )
+
+        self.eta_ratio = Var(
+            self.flowsheet().config.time,
+            units=pyunits.dimensionless,
+            bounds=(0, 1),
+            doc="Ratio between the true efficiency and the best efficiency due to variable operation",
+        )
+        self.electricity = Var(
+            self.flowsheet().config.time,
+            units=pyunits.kW,
+            bounds=(0, None),
+            doc="Electricity for low pressure pump",
+        )
+
+        self.applied_pressure = Var(
+            self.flowsheet().config.time,
+            units=pyunits.bar,
+            bounds=(0, None),
+            doc="Applied pressure",
+        )
+
+        self._fixed_perf_vars.append(self.lift_height)
+        self._fixed_perf_vars.append(self.eta_pump)
+        self._fixed_perf_vars.append(self.eta_motor)
+
+        @self.Constraint(
+            self.flowsheet().time, doc="Constraint for pump applied pressure"
+        )
+        def applied_pressure_constraint(b, t):
+            return b.applied_pressure[t] == pyunits.convert(
+                b.lift_height
+                * b.properties[t].dens_mass
+                * Constants.acceleration_gravity,
+                to_units=pyunits.bar,
+            )
+
+        @self.Constraint(
+            self.flowsheet().time, doc="Constraint for variable pump flowrate ratio"
+        )
+        def flow_ratio_constraint(b, t):
+            return b.flow_ratio[t] == b.properties[t].flow_vol / pyunits.convert(
+                b.flow_bep, to_units=pyunits.m**3 / pyunits.s
+            )
+
+        @self.Constraint(
+            self.flowsheet().time, doc="Constraint for variable pump efficiency"
+        )
+        def eta_ratio_constraint(b, t):
+            # if the flow is too far from the design point, the efficiency is set to 40%
+            if b.flow_ratio[t] < 0.6 or b.flow_ratio[t] > 1.4:
+                return b.eta_ratio[t] == 0.4
+            # otherwise, use the correlation
+            else:
+                return b.eta_ratio[t] == (
+                    -0.995 * b.flow_ratio[t] ** 2 + 1.977 * b.flow_ratio[t] + 0.025
+                )
+
+        @self.Constraint(
+            self.flowsheet().time, doc="Constraint for actual pump energy consumption"
+        )
+        def electricity_consumption(b, t):
+            return b.electricity[t] == pyunits.convert(
+                b.lift_height
+                * b.properties[t].flow_vol
+                * b.properties[t].dens_mass
+                * Constants.acceleration_gravity
+                / (b.eta_pump * b.eta_motor * b.eta_ratio[t]),
```

Comment:
We'll want to think about this, but additional details down the line could include part-load motor efficiency and _maybe_ the same for a VFD if we add that in.

Reply:
In my previous work on, I just assumed the variable efficiency to be represented by the pump- the simplifying assumption being that the pump efficiency represented the overall efficiency. That is, I didn't directly account for motor or VFD efficiency. So accounting for these effects would be an improvement.

Reply:
My modeling approach has been to just have a single pump efficiency that is the combined pump and motor efficiency. In some papers I would give both values - but I'd model the efficiency as one variable, but I think the key thing we need to think about is how do we expect users to interact with these parameters. For me, if someone wanted to vary the efficiency of the pump for a sensitivity they'd just want to vary one value rather than both, and I don't think anyone would really want to vary the two efficiencies separately - which leads me to think we should just combine them.

Reply:
A lot of people like to vary the parameters separately - not so much for sensitivity studies but for tracking different contributions. E.g. we have one collaboration in IDAES where the compressors will have 3 efficiencies (thermodynamic, mechanical and electric).

Reply:
I would also opt for having separate efficiencies. If we were to ever to go to the highest level of detail for flexible pump operation. While assuming the pump efficiency to be representative of the overall efficiency is a valid assumption, in the future, what we'd (I'd) really want is to have  pump, motor, and VFD efficiencies, all with part-load performance, but no need to go there yet.